core/dbt/events/functions.py

```diff
@@ -168,11 +168,7 @@ def msg_to_dict(msg: EventMsg) -> dict:
 
 
 def warn_or_error(event, node=None):
-    # TODO: resolve this circular import when flags.WARN_ERROR_OPTIONS is WarnErrorOptions type via click CLI.
-    from dbt.helper_types import WarnErrorOptions
-
-    warn_error_options = WarnErrorOptions.from_yaml_string(flags.WARN_ERROR_OPTIONS)
-    if flags.WARN_ERROR or warn_error_options.includes(type(event).__name__):
```

Comment:
@iknox-fa - click's [type conversion](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/cli/params.py#L374) enables some nice cleanup here and [here](https://github.com/dbt-labs/dbt-core/pull/6762/files#diff-1f46b05ee5e8edb06d10f6a5104795ccda0d1c2c458a203db4914d68115978b5L126-L141)! ✨